src/compile/Component.ts

```diff
@@ -1050,6 +1050,7 @@ export default class Component {
 				this.reactive_declaration_nodes.add(node);
 
 				const assignees = new Set();
+				const assignee_nodes = new Set();
```

Comment:
Keeping track of the `assignee_nodes` when walking and skipping over those seems like a good way to do this :+1: